include/lsst/afw/typehandling/PolymorphicValue.h

```diff
@@ -137,7 +137,8 @@ class PolymorphicValue final {
     std::size_t hash_value() const;
 
 private:
-    std::unique_ptr<Storable> _value;
+    // unique_ptr would be more appropriate, but Storable::cloneStorable must return shared_ptr
+    std::shared_ptr<Storable> _value;
```

Comment:
I don't understand your first point. How does making a factory method return `shared_ptr` "allow" `make_shared`? Can't you call `make_shared` for any class that has a public constructor?

Re: points 2 and 3, I don't think `Psf` is the problem. It's simply the first time we've tried to retrofit `Storable` onto an existing class hierarchy. I already rejected the manual lambda idea on the grounds that it would make it hard to use `Storable` correctly -- anybody trying to make another cloneable class storable would need to know to repeat the trick.

I don't think the fourth point is relevant because we don't want to transfer ownership from Python (a C++ function that takes `unique_ptr` input would have unintuitive side effects anyway). Having a `unique_ptr` return value corresponds to transferring ownership *to* Python, which we do often.

Reply:
On point (1), I just mean that if you define `clone` to return `unique_ptr`, but always in practice convert that `unique_ptr` into a `shared_ptr`, you've missed out on the optimization that `make_shared` provides, while if you define `clone` to return `shared_ptr`, you can implement it via `make_shared`.